ablySpec/RealtimeClientChannel.swift

```diff
@@ -609,6 +609,88 @@ class RealtimeClientChannel: QuickSpec {
+                    defer { client2.close() }
+
+                    let channel1 = client1.channels.get("test")
+                    let channel2 = client2.channels.get("test")
+
+                    class Test {
+                        static var counter = 0
+                        private init() {}
+                    }
+
+                    channel1.subscribe { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Test.counter += 1
+
+                        channel1.unsubscribe()
+                    }
+
+                    channel2.subscribe { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Test.counter += 1
+                    }
+
+                    channel1.publish(nil, data: "message")
+                    channel1.publish(nil, data: "message")
+
+                    expect(Test.counter).toEventually(equal(3), timeout: testTimeout)
+                }
+
+                // RTL8b
+                it("with a single name argument unsubscribes the provided listener if previously subscribed with a name-specific subscription") {
+                    let options = AblyTests.commonAppSetup()
+                    let client1 = ARTRealtime(options: options)
+                    defer { client1.close() }
+                    let client2 = ARTRealtime(options: options)
+                    defer { client2.close() }
+
+                    let channel1 = client1.channels.get("test")
+                    let channel2 = client2.channels.get("test")
+
+                    class Counter {
+                        static var eventA = 0
+                        static var eventB = 0
+                        private init() {}
+                    }
+
+                    let eventAListener = channel1.subscribe("eventA") { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Counter.eventA += 1
+                    }
+
+                    channel1.subscribe("eventB") { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Counter.eventB += 1
+                    }
+
+                    channel2.subscribe("eventA") { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Counter.eventA += 1
+
+                        channel1.unsubscribe("eventA", listener: eventAListener)
+                    }
+
+                    channel1.publish("eventA", data: "message")
+                    channel1.publish("eventB", data: "message")
+                    channel2.publish("eventA", data: "message")
+                    channel2.publish("eventB", data: "message")
+
+                    expect(Counter.eventA).toEventually(equal(3), timeout: testTimeout)
+                    expect(Counter.eventB).toEventually(equal(2), timeout: testTimeout)
```

Comment:
As above, why complicate this with two channels.  Why not simply do what I described above with one channel, publish eventA and eventB, receive both via subscribers, unsubscribe eventA subscriber, publish both, check that only eventB is received.